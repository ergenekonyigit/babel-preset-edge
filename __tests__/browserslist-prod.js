import buildPreset from "../src"
import { fixtures, titles, check } from "./core"

process.env.NODE_ENV = "production"

describe("Browserslist Production", () => {
  const options = buildPreset(null, {
    target: "browser",
    sourceMaps: false,
    env: "production"
  })

  fixtures.forEach((fileName, index) => {
    test(titles[index], () => check(fileName, options))
  })
})
