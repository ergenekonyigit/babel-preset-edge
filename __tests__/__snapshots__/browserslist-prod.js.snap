// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Browserslist Production comments 1`] = `
"import _camelCase from \\"lodash/camelCase\\";

// Line Comment
function translate(a) {
  return _camelCase(
  /* inline Comment */
  a);
}"
`;

exports[`Browserslist Production deadcode 1`] = `
"var STATIC = !1;
!1, !1;"
`;

exports[`Browserslist Production es3: literal 1`] = `"foo[\\"catch\\"];"`;

exports[`Browserslist Production es3: member 1`] = `
"var foo = {
  catch: function _catch() {}
};"
`;

exports[`Browserslist Production es2015: classes 1`] = `
"export var Component =
/*#__PURE__*/
function () {
  function a() {}

  var b = a.prototype;
  return b.main = function main() {}, a;
}();"
`;

exports[`Browserslist Production es2015: for_off 1`] = `
"import \\"core-js/modules/es7.symbol.async-iterator\\";
import \\"core-js/modules/es6.symbol\\";
import \\"core-js/modules/web.dom.iterable\\";

function test() {
  for (var a = tasks, b = Array.isArray(a), c = 0, a = b ? a : a[Symbol.iterator]();;) {
    var d;

    if (b) {
      if (c >= a.length) break;
      d = a[c++];
    } else {
      if (c = a.next(), c.done) break;
      d = c.value;
    }

    try {
      executeCommands();
    } catch (a) {
      console.error(\\"Error\\");
    }
  }
}"
`;

exports[`Browserslist Production es2015: generator 1`] = `
"function* idMaker() {
  for (var a = 0;;) yield a++;
}"
`;

exports[`Browserslist Production es2015: set 1`] = `
"import \\"core-js/modules/es6.set\\";
new Set([1, 2, 3]);"
`;

exports[`Browserslist Production es2016: exponentiation 1`] = `"var result = 8;"`;

exports[`Browserslist Production esnext: async 1`] = `
"function wait() {
  return new Promise(function (a, b) {
    return Promise.resolve(loadData()).then(function () {
      try {
        return a();
      } catch (a) {
        return b(a);
      }
    }, b);
  });
}"
`;

exports[`Browserslist Production esnext: async_loop 1`] = `
"import \\"core-js/modules/es7.symbol.async-iterator\\";
import \\"core-js/modules/es6.symbol\\";
import \\"core-js/modules/web.dom.iterable\\";

function executeTasks(a) {
  return new Promise(function (b, c) {
    function d() {
      if (g) {
        if (h >= f.length) return [1];
        i = f[h++];
      } else {
        if (h = f.next(), h.done) return [1];
        i = h.value;
      }

      var a = function () {
        try {
          return d;
        } catch (a) {
          return c(a);
        }
      },
          b = function () {
        try {
          return console.error(\\"Error\\"), a();
        } catch (a) {
          return c(a);
        }
      };

      try {
        return Promise.resolve(executeCommands()).then(function () {
          try {
            return a();
          } catch (a) {
            return b(a);
          }
        }, b);
      } catch (a) {
        b(a);
      }
    }

    function e() {
      return b();
    }

    var f, g, h, i;
    f = a, g = Array.isArray(f), h = 0, f = g ? f : f[Symbol.iterator]();
    var j;
    return (j = function (a) {
      for (; a;) {
        if (a.then) return void a.then(j, c);

        try {
          if (a.pop) {
            if (a.length) return a.pop() ? e.call(this) : a;
            a = d;
          } else a = a.call(this);
        } catch (a) {
          return c(a);
        }
      }
    }.bind(this))(d);
  });
}"
`;

exports[`Browserslist Production esnext: class properties 1`] = `
"import _defineProperty from \\"@babel/runtime/helpers/builtin/es6/defineProperty\\";
export var Component = function () {
  _defineProperty(this, \\"onClick\\", function () {});
};"
`;

exports[`Browserslist Production esnext: dynamic import 1`] = `
"import \\"core-js/modules/es6.promise\\";
import \\"core-js/modules/web.dom.iterable\\";
import _path from \\"path\\";
import _importCss from \\"babel-plugin-universal-import/importCss\\";
import _universalImport from \\"babel-plugin-universal-import/universalImport\\";

_universalImport({
  id: \\"lodash\\",
  file: \\"./__tests__/__fixtures__/esnext_dynamic-import.js\\",
  load: function load() {
    return Promise.all([import(
    /* webpackChunkName: 'lodash' */
    \\"lodash\\"), _importCss(\\"lodash\\", {})]).then(function (a) {
      return a[0];
    });
  },
  path: function path() {
    return _path.join(__dirname, \\"lodash\\");
  },
  resolve: function resolve() {
    return require.resolveWeak(\\"lodash\\");
  },
  chunkName: function chunkName() {
    return \\"lodash\\";
  }
}).then(function (a) {
  console.log(a.camelCase(\\"hello world\\"));
});"
`;

exports[`Browserslist Production esnext: import 1`] = `
"import _camelCase from \\"lodash/camelCase\\";
console.log(_camelCase(\\"hello world\\"));"
`;

exports[`Browserslist Production esnext: object spread 1`] = `
"import _objectSpread from \\"@babel/runtime/helpers/builtin/es6/objectSpread\\";

var original = {
  bar: 10
},
    variant = _objectSpread({
  foo: 1
}, original);"
`;

exports[`Browserslist Production lodash: async 1`] = `
"import _parallel from \\"async/parallel\\";

_parallel([]);"
`;

exports[`Browserslist Production lodash: lodash 1`] = `
"import _camelCase from \\"lodash/camelCase\\";

_camelCase(\\"hello world\\");"
`;

exports[`Browserslist Production lodash: recompose 1`] = `
"import _compose from \\"recompose/compose\\";

_compose([]);"
`;

exports[`Browserslist Production react: jsx 1`] = `
"var _ref =
/*#__PURE__*/
React.createElement(\\"h1\\", null, \\"Hello\\");

function render() {
  return _ref;
}"
`;

exports[`Browserslist Production react: jsx spread 1`] = `
"import \\"core-js/modules/es6.object.assign\\";

function render(a) {
  return React.createElement(\\"h1\\", Object.assign({
    className: \\"large\\"
  }, a), \\"Hello\\");
}"
`;

exports[`Browserslist Production react: proptypes 1`] = `
"import _inheritsLoose from \\"@babel/runtime/helpers/builtin/es6/inheritsLoose\\";
import React from \\"react\\";

var MyComponent =
/*#__PURE__*/
function (a) {
  function b() {
    return a.apply(this, arguments) || this;
  }

  var c = b.prototype;
  return c.render = function render() {
    return React.createElement(\\"div\\", null, this.props.userName);
  }, _inheritsLoose(b, a), b;
}(React.Component);"
`;
